### PR TITLE
Add basic support for Caliper

### PR DESCRIPTION
**Pull request by daboehme**

### CODE REVIEW

**Comment by markcmiller86** [comment]
ping @daboehme

**Comment by daboehme** [comment]
This is ready to go from my side. Any questions @markcmiller86 ?

**Comment by markcmiller86** [comment]
Yes, see above. Are they not visible to you?

**Comment by daboehme** [comment]
The "ping daboehme" comment is the only thing I see. Strange.

**Comment by markcmiller86** [comment]
Sheesh. Sorry. I had commented a month ago but I guess forgot to hit the "Start Review" button so it just sat there. My mistake.

File: INSTALLING.md [hunk 1]
```diff
@@ -16,6 +16,7 @@
     - Build TyphonIO Plugin:  -DENABLE_TYPHONIO_PLUGIN=ON -DWITH_TYPHONIO_PREFIX=[path to typhonio]
     - Build PDB Plugin:       -DENABLE_PBD_PLUGIN=ON
     - Build Exodus Plugin:    -DENABLE_EXODUS_PLUGIN=ON -DWITH_EXODUS_PREFIX=[path to exodus]
+    - Caliper support:        -DENABLE_CALIPER=ON -Dcaliper_DIR=[caliper-install-dir]/share/cmake/caliper
```
[the remark this answers is not in the document]
Comment (daboehme) [reply INSTALLING.md]: This variable name has to match Caliper's CMake project name, which is lower-case "caliper". It probably would have been better to at least use "Caliper", but it's a bit too late to change now.

File: macsio/macsio_timing.c [hunk 2]
```diff
@@ -192,6 +196,10 @@ typedef struct _timerInfo_t
     char __file__[32];               /**< Source file name for StartTimer call */
     char label[64];                  /**< User defined label given to the timer */
 
+#ifdef HAVE_CALIPER
+    cali_id_t attr;                  /**< Caliper attribute id for the timer */
+    cali_id_t iter_attr;             /**< Caliper attribute id for the iteration */
+#endif
```
[the remark this answers is not in the document]
Comment (daboehme) [reply macsio/macsio_timing.c]: Oh, I didn't catch that, even though there is even a comment on the data structure mentioning that. Guess I should learn how to read. I'll get that fixed.

**Comment by daboehme** [comment]
No worries :) I'll fix the MPI datatype issue. Thanks for catching that.

**Comment by daboehme** [comment]
Hi @markcmiller86 ,

The timer info struct issue should be fixed now. I took the Caliper attributes out of there and moved them into their own struct. Caliper has its own cross-process aggregation logic.

**Comment by markcmiller86** [comment]
Thanks so much @daboehme

**Pull request merged**